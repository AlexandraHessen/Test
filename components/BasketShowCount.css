.BasketShowCount  .fa-shopping-cart{
    color: #fff;
    font-size: 1.7em;
    position: relative
}

.BasketShowCount  .ButtonToBasket{
    border: solid 2px #8ccb68;
    font-size: 1em;
    padding: 8px 13px;
    border-radius: 5px;
    background-color: transparent;
    cursor: pointer;
    position: relative

}

.BasketShowCount .CountProductsInBasket{
    position: absolute;
    top: -8px;
    right: -8px;
    width: 25px;
    height: 25px;
    border-radius: 50%;
    background-color: #357A10;
    color: white;
    text-align: center;
    line-height: 25px;
    font-size: 0.9em;
    font-weight: bold;
}

@media all and (max-width: 500px){
    .BasketShowCount  .ButtonToBasket{
        padding: 6px 10px;
    }

    .BasketShowCount .CountProductsInBasket{

        width: 20px;
        height: 20px;

        line-height: 25px;
    }

}