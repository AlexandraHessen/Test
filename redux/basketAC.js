// ---------------- ActionCreator ----------------//
const ADD_PRODUCT = "ADD_PRODUCT"
const DEL_PRODUCT = "DEL_PRODUCT"
const CLEAR_BASKET = "CLEAR_BASKET" //пользователь сам очищает корзину при желании

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_product = function(productId, objProductInfo) {
    return {
      type: ADD_PRODUCT,
      productId: productId,
      objProductInfo: objProductInfo
    };
  }
  
  const del_product = function(productId) {
    return {
      type: DEL_PRODUCT,
      productId: productId,
    };
  }

  const clear_basket = function() {
    return {
      type: CLEAR_BASKET
    };
  }

export {
    add_product, ADD_PRODUCT,
    del_product, DEL_PRODUCT,
    clear_basket, CLEAR_BASKET,
  }