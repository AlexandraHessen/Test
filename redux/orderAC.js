// ---------------- ActionCreator ----------------//
const ADD_ORDER = "ADD_ORDER"

//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator

const add_order = function(objCustomerInfo, objOrderInfo) {
    return {
      type: ADD_ORDER,
      objCustomerInfo: objCustomerInfo,
      objOrderInfo: objOrderInfo
    };
  }

export {
    add_order, ADD_ORDER,
  }