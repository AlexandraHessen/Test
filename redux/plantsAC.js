// ---------------- ActionCreator ----------------//
const PLANTS_LOADING='PLANTS_LOADING';
const PLANTS_ERROR='PLANTS_ERROR';
const PLANTS_SET='PLANTS_SET';
//название action type

//функция для dispatch (из файла CounterButton) которая возвращает хэш с action type 
// называется ActionCreator
const plantsLoadingAC=function() {
  return {
    type: PLANTS_LOADING,
  };
}

const plantsErrorAC=function() {
  return {
    type: PLANTS_ERROR,
  };
}

const plantsSetAC=function(plants) {
  return {
    type: PLANTS_SET,
    plants:plants,
  };
}

export {
  plantsLoadingAC,PLANTS_LOADING,
  plantsErrorAC,PLANTS_ERROR,
  plantsSetAC,PLANTS_SET,
}
