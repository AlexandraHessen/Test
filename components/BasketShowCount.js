"use strict";

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import {connect} from 'react-redux';

import './BasketShowCount.css'

class BasketShowCount extends React.Component{
    static propTypes = {
        basket: PropTypes.object.isRequired // пришло из Redux (для отображения количества товаров в корзине)
                                            // в ответ ничего не передаем 
                                            // ключ productsInBasket (this.props.basket.productsInBasket)
    };
    
    render(){
        let countProductsInBasket=Object.keys(this.props.basket.productsInBasket).length;
        
        return(

        <NavLink to="/basket"  className="BasketShowCount">
            <button type="button" className="ButtonToBasket">
                <i className="fas fa-shopping-cart"></i>
                <div className="CountProductsInBasket">
                    {countProductsInBasket}
                </div>
            </button>
        </NavLink>

        )
    }
}

const mapStateToProps = function (state) {
    return {
      // весь раздел Redux state под именем basket будет доступен
      // данному компоненту как this.props.basket
      basket: state.basket,
    };
  };
  
  export default connect(mapStateToProps)(BasketShowCount);
  // экспортируем не этот класс BasketShowCount а уже обвернутый

