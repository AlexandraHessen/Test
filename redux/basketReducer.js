// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_PRODUCT, DEL_PRODUCT, CLEAR_BASKET } from './basketAC'; //название action type


const initState={ // начальный state
    productsInBasket: {}
    //КЛЮЧИ productId, objProductInfo
    // 1: {code: 1, name: 'Замиокулькас'}

//   status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
//   data: null, //даные которые мы загрузили по сети

}

function basketReducer(state=initState,action) {
  switch (action.type) {

    case ADD_PRODUCT: {
      console.log('action:',action);
      console.log('state до обработки редьюсером:',state);
      let newState={...state, //старый state
        productsInBasket:{...state.productsInBasket, //заменить старые продукты на теже +
          // ПРИЛЕТИТ какой продукт (productid) изменить и насколько(addvalue)
          [action.productId]:action.objProductInfo //+
          //[action.counterid] - В нужном продукте 
          // state.productsInBasket[action.productid] - его старое значение 
          // action.addvalue - изменить на прилетевшее
        }
      };
      console.log('state после обработки редьюсером:',newState);
      return newState;
    }

    case DEL_PRODUCT: {
      let newState = {...state};
      delete newState.productsInBasket[action.productId];
      return newState;
    }

    case CLEAR_BASKET:{
      let newState = {...state};
      newState.productsInBasket={};
      return newState;
    }

    // case PLANTS_ERROR: {
    //   let newState={
    //     status:2,
    //     data:null,
    //   };
    //   return newState;
    // }

    // case PLANTS_SET: {
    //   let newState={
    //     status:3,
    //     data:action.plants, //загруженные данные
    //   };
    //   return newState;
    // }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default basketReducer;
