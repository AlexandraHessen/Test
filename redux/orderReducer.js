// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { ADD_ORDER } from './orderAC'; //название action type

const initState={ // начальный state
    orders: []
}

function orderReducer(state=initState,action) {
  switch (action.type) {
    case ADD_ORDER: {
      let newOrder=[...state.orders]
      newOrder.push( { objCustomerInfo: action.objCustomerInfo, products: action.objOrderInfo } );
      let newState={...state, orders: newOrder};

      return newState;
    }

              //обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default orderReducer;