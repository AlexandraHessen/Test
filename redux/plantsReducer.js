// ---------------- САМА ФУНКЦИЯ REDUСER  ----------------//
// REDUCER не делает запрос в Ajax, логика должна быть в компоненте
import { PLANTS_LOADING, PLANTS_ERROR, PLANTS_SET } from './plantsAC'; //название action type


const initState={ // начальный state

  status: 0, // 0 - ничего не началось, 1 - идёт загрузка, 2 - была ошибка, 3 - данные загружены
  data: null, //даные которые мы загрузили по сети

}

function plantsReducer(state=initState,action) {
  switch (action.type) {

    case PLANTS_LOADING: {
      let newState={
        status:1,
        data:null,
      };
      return newState;
    }

    case PLANTS_ERROR: {
      let newState={
        status:2,
        data:null,
      };
      return newState;
    }

    case PLANTS_SET: {
      let newState={
        status:3,
        data:action.plants, //загруженные данные
      };
      return newState;
    }
    
//обязатнльно пишем на тот случай если пришел action который не относится к этому reducery
    default:
      return state;
  }
}

export default plantsReducer;
