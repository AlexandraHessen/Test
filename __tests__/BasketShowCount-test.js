"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import BasketShowCount  from '../components/BasketShowCount';
                    //импортируем файл который будем тестировать

let info = 	{
    "code": 1,
    "name": "Замиокулькас",
    "category": "deciduous",
    "price": 265,
    "description": "Полив по мере полного просыхания земляного кома, раз в 10-15 дней, в опрыскивании не нуждается. Чрезмерный полив замиокулькасов может приводить к пожелтению листьев.",
    "imgUrl": "/img/img_9241_1.jpg"
}

let store=createStore(combinedReducer);

test('BasketShowCount ', () => {
                            // создаём тестовую версию компонента который будем тестировать
  const component = renderer.create(
  <Provider store={store}>
    <BrowserRouter>
      <BasketShowCount  basket = {info}/>
    </BrowserRouter>
  </Provider>

  );

                            // получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
});
