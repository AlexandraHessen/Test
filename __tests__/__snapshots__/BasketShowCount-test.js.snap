// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasketShowCount  1`] = `
<a
  aria-current={false}
  className="BasketShowCount"
  href="/basket"
  onClick={[Function]}
  style={undefined}
>
  <button
    className="ButtonToBasket"
    type="button"
  >
    <i
      className="fas fa-shopping-cart"
    />
    <div
      className="CountProductsInBasket"
    >
      0
    </div>
  </button>
</a>
`;
